polygeist/include/mlir/Dialect/Polygeist/IR/PolygeistTypes.td

```diff
@@ -32,8 +32,14 @@ def Polygeist_StructType :
     Type used to represent a collection of data members together in memory.
   }];
   let parameters = (ins ArrayRefParameter<"mlir::Type">:$body,
-                        OptionalParameter<"std::optional<StringAttr>">:$name);
-  let assemblyFormat = "`<` ($name^`,` )? `(` $body `)` `>`";
+                        OptionalParameter<"std::optional<StringAttr>">:$name,
+                        DefaultValuedParameter<"bool", "false">:$isPacked);
+  let builders = [
+    TypeBuilder<(ins "llvm::ArrayRef<mlir::Type>":$body),[{
+      return $_get($_ctxt, body, std::nullopt, false);
+    }]>
+  ];
+  let assemblyFormat = "`<` ($name^ `,` ` `)? (`isPacked` `=` $isPacked^ ` `)? `(` $body `)` `>`";
```

Comment:
Can we do have `packed` printed instead of `isPacked = true`?

Reply:
Do you know how to do that? Do I need to create custom printer for that?

Reply:
Maybe this is applicable here: https://mlir.llvm.org/docs/DefiningDialects/Operations/#optional-groups

Reply:
Try:
```
(`packed` $isPacked^)?
```
 in place of 
 ```
 (`isPacked` `=` $isPacked^ ` `)?
 ```

Reply:
Doesn't work:
```
polygeist/test/polygeist-opt/polygeist-types-to-llvm.mlir:15 offset :4:64: error: failed to parse Polygeist_StructType parameter 'isPacked' which is to be a `bool`
func.func @test_struct(%arg0: !polygeist.struct<"name", packed (memref<i32>, i32)>) {
```

Reply:
This is fine in that case. LGTM